source/array_list.c

```diff
@@ -137,7 +139,7 @@ int aws_array_list_pop_back(struct aws_array_list *list) {
 }
 
 void aws_array_list_clear(struct aws_array_list *list) {
-    if (list->length > 0) {
+    if (list->data && list->length > 0) {
```

Comment:
you don't have to change this line, length can't be >0 when data is NULL

Reply:
done